src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/Metafile.Windows.COMWrappers.cs

```diff
@@ -0,0 +1,129 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Drawing.Internal;
+using System.Runtime.InteropServices;
+using Gdip = System.Drawing.SafeNativeMethods.Gdip;
+
+namespace System.Drawing.Imaging
+{
+    public sealed partial class Metafile : Image
+    {
+        private static unsafe IntPtr CreateGdipMetafileFromStream(GPStream stream)
+        {
+            DrawingComWrappers.Instance.GetIStreamInterfaces(stream, out IntPtr streamWrapperPtr, out IntPtr streamPtr);
```

Comment:
This whole file can be shared between the COM wrappers and downlevel implementations if you provide downlevel implementation for the DrawingComWrappers. The downlevel implementation can look like this:

```csharp
class DrawingComWrappers
{
    static IntPtr GetIStreamInterfaces(GPStream stream)
    {
        return Marshal.GetComInterfaceForObject<GPStream, IStream>(stream);
    }
}
```
(This assumes my other feedback about GetIStreamInterfaces. Also, the method is static.)